src/test/fuzz/versionbits.cpp

```diff
@@ -0,0 +1,343 @@
+    const int m_period = 0;
+    const int m_threshold = 0;
+    const int m_bit = 0;
+
+    TestConditionChecker(int64_t begin, int64_t end, int period, int threshold, int bit) : m_begin{begin}, m_end{end}, m_period{period}, m_threshold{threshold}, m_bit{bit}
+    {
+        assert(m_period > 0);
+        assert(0 <= m_threshold && m_threshold <= m_period);
+        assert(0 <= m_bit && m_bit <= 32 && m_bit < VERSIONBITS_NUM_BITS);
+    }
+
+    virtual bool Condition(const CBlockIndex* pindex, const Consensus::Params& params) const override { return Condition(pindex->nVersion); }
+    virtual int64_t BeginTime(const Consensus::Params& params) const override { return m_begin; }
+    virtual int64_t EndTime(const Consensus::Params& params) const override { return m_end; }
+    virtual int Period(const Consensus::Params& params) const override { return m_period; }
+    virtual int Threshold(const Consensus::Params& params) const override { return m_threshold; }
+
+    ThresholdState GetStateFor(const CBlockIndex* pindexPrev) const { return AbstractThresholdConditionChecker::GetStateFor(pindexPrev, dummy_params, m_cache); }
+    int GetStateSinceHeightFor(const CBlockIndex* pindexPrev) const { return AbstractThresholdConditionChecker::GetStateSinceHeightFor(pindexPrev, dummy_params, m_cache); }
+    BIP9Stats GetStateStatisticsFor(const CBlockIndex* pindexPrev) const { return AbstractThresholdConditionChecker::GetStateStatisticsFor(pindexPrev, dummy_params); }
+
+    bool Condition(int64_t version) const
+    {
+        return ((version >> m_bit) & 1) != 0 && (version & VERSIONBITS_TOP_MASK) == VERSIONBITS_TOP_BITS;
+    }
+
+    bool Condition(const CBlockIndex* pindex) const { return Condition(pindex->nVersion); }
+};
+
+/** Track blocks mined for test */
+class Blocks
+{
+private:
+    std::vector<CBlockIndex*> blocks;
+    const uint32_t m_start_time;
+    const int32_t m_signal;
+    const int32_t m_no_signal;
+
+public:
+    Blocks(uint32_t start_time, int32_t signal, int32_t no_signal) : m_start_time{start_time}, m_signal{signal}, m_no_signal{no_signal} { }
+
+    ~Blocks()
+    {
+        for (auto& v : blocks) {
+            delete v;
+        }
+        blocks.clear();
+    }
+
+    size_t size() const { return blocks.size(); }
+
+    CBlockIndex* tip() const
+    {
+        return blocks.empty() ? nullptr : blocks.back();
+    }
+
+    CBlockIndex* mine_block(bool signal)
+    {
+        CBlockHeader header;
+        header.nVersion = signal ? m_signal : m_no_signal;
+        header.nTime = m_start_time + blocks.size() * 600;
+        header.nBits = 0x1d00ffff;
+
+        CBlockIndex* current_block = new CBlockIndex{header};
+        current_block->pprev = tip();
+        current_block->nHeight = blocks.size();
+        current_block->BuildSkip();
+
+        blocks.push_back(current_block);
+
+        return current_block;
+    }
+};
+
+void initialize()
+{
+    return;
+}
+} // namespace
+
+FUZZ_TARGET_INIT(versionbits, initialize)
```

Comment:
style-nit: Can be shorter

```suggestion
FUZZ_TARGET(versionbits)
```